src/domain/safe/entities/transfer.entity.ts

```diff
@@ -7,14 +7,30 @@ export interface Transfer {
   transferId: string;
 }
 
+const hasTokenAddress = (
+  transfer: Transfer,
+): transfer is Transfer & {
+  tokenAddress: (ERC20Transfer | ERC721Transfer)['tokenAddress'];
+} => {
+  return 'tokenAddress' in transfer && transfer.tokenAddress != null;
+};
+
+const hasValue = (
+  transfer: Transfer,
+): transfer is Transfer & {
+  value: (ERC20Transfer | NativeTokenTransfer)['value'];
+} => {
+  return 'value' in transfer && transfer.value != null;
+};
+
 export interface ERC20Transfer extends Transfer {
   tokenAddress: string;
   value: string;
 }
 
 export function isERC20Transfer(transfer: Transfer): transfer is ERC20Transfer {
-  const erc20Transfer = transfer as ERC20Transfer;
-  return erc20Transfer.tokenAddress != null && erc20Transfer.value != null;
+  const hasValue = 'value' in transfer && transfer.value != null;
```

Comment:
I'm a bit confused about this line. Isn't `hasValue` already declared as a function above?

Reply:
It is, I missed this one. I've remove it in 7cc62fa26e597a277f5adc5a0bda4ed3cba0e0fc.